### PR TITLE
Update answer and studio to gpt4-turbo 24k

### PR DESCRIPTION
**Pull request by rmuller-ml**
- We separate the model from agent and answer to have different headrooms
- We don't use the full 128k context window (only 24k)

### CODE REVIEW

File: server/bleep/src/agent.rs [hunk 1]
```diff
@@ -57,7 +57,8 @@ pub struct Agent {
     pub thread_id: uuid::Uuid,
     pub query_id: uuid::Uuid,
 
-    pub model: model::AnswerModel,
+    pub answer_model: model::LLMModel,
+    pub agent_model: model::LLMModel,
```
Comment (nerdypepper) [inline server/bleep/src/agent.rs:61]: i think its confusing that `Agent` has, within itself, an `agent_model` field, can we call this something else? such as `decsicion_model` or `reasoning_model`?

File: server/bleep/src/webserver/answer.rs [hunk 2]
```diff
@@ -69,8 +69,10 @@ pub(super) async fn vote(
 pub struct Answer {
     pub q: String,
     pub repo_ref: RepoRef,
-    #[serde(default = "default_model")]
-    pub model: agent::model::AnswerModel,
+    #[serde(default = "default_answer_model")]
+    pub answer_model: agent::model::LLMModel,
+    #[serde(default = "default_agent_model")]
+    pub agent_model: agent::model::LLMModel,
```
Comment (nerdypepper) [inline server/bleep/src/webserver/answer.rs:75]: i believe this is a breaking change to the API. the FE currently this parameter in `model`, but the name has been changed to `answer_model`. here is what i think we should do:

* so we can either make the corresponding change on the FE here (cc @anastasiya1155)
* call it `answer_model` and use `#[serde(rename = "model")]`
* revert to old name   

side note: do we want to be able to configure `agent_model` with request parameter??

**Review by nerdypepper — CHANGES_REQUESTED** [review]

File: client/src/components/Chat/index.tsx [hunk 3]
```diff
@@ -94,8 +94,10 @@ const Chat = () => {
               queryIdToEdit ? `&parent_query_id=${queryIdToEdit}` : ''
             }`
           : ''
-      }&model=${
-        preferredAnswerSpeed === 'normal' ? 'gpt-4' : 'gpt-3.5-turbo-finetuned'
+      }&answer_model=${
```
Comment (rmuller-ml) [inline client/src/components/Chat/index.tsx:97]: Making it compatible with Answer struct interface change 'model' => 'answer_model'

File: server/bleep/src/webserver/answer.rs (hunk 2 above)
Comment (rmuller-ml) [reply server/bleep/src/webserver/answer.rs:75]: I changed in the FE from model to answer_model (see comment in client/src/components/Chat/index.tsx)

File: server/bleep/src/agent.rs (hunk 1 above)
Comment (rmuller-ml) [reply server/bleep/src/agent.rs:61]: I think `agent_model` is fine.
The abstraction is:  the user queries to get an `answer` and we have an `answer_model` used in the tool `answer`, to get the the correct context to answer the query we start an `agent` which has its own `agent_model`

Third opinion @ggordonhall ?

**Review by nerdypepper — APPROVED** [review]

**Pull request merged**